abb_hardware_interface/src/abb_system_position_only.cpp

```diff
@@ -12,243 +12,199 @@
-          abb::egm::wrapper::Status_RAPIDExecutionState_RAPID_RUNNING;
-      }
-    } else {
-      RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "Not Connected to Robot...");
+  RCLCPP_INFO(LOGGER, "Connecting to robot...");
+  while (rclcpp::ok() && ++counter < NUM_CONNECTION_TRIES) {
+    // Wait for a message on any of the configured EGM channels.
+    if (egm_manager_->waitForMessage(500)) {
+      RCLCPP_INFO(LOGGER, "Connected to robot");
+      break;
+    }
+    
+    RCLCPP_INFO(LOGGER, "Not connected to robot...");
+    if (counter == NUM_CONNECTION_TRIES) {
+      RCLCPP_ERROR(LOGGER, "Failed to connect to robot");
+      return CallbackReturn::ERROR;
     }
     rclcpp::sleep_for(500ms);
   }
-  if (wait) {  // if still not in the right state, exit
-    RCLCPP_FATAL(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "Could NOT Connect to Robot");
-    return CallbackReturn::ERROR;
-  }
-  RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "CONNECTED!");
-
-  // SET INITIAL STATE TO CURRENT ANGLES, error if something isn't right
-  if (!egm_interface_->waitForMessage(500)) {  // in [ms]
-    RCLCPP_FATAL(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "NO Message Received");
-    return CallbackReturn::ERROR;
-  }
-
-  egm_interface_->read(&input_);
-  sequence_number_ = input_.header().sequence_number();
-
-  if (sequence_number_ <= 0) {
-    RCLCPP_FATAL(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "Sequence number < 0");
-    return CallbackReturn::ERROR;
-  }
-
-  output_.Clear();
-  output_pos_.Clear();
-  current_positions_.CopyFrom(input_.feedback().robot().joints().position());
-
-  if (static_cast<size_t>(current_positions_.values_size()) != hw_states_.size()) {
-    RCLCPP_FATAL(
-      rclcpp::get_logger("ABBSystemPositionOnlyHardware"),
-      "Robot does not match expected number of joints");
-    return CallbackReturn::ERROR;
-  }
-
-  output_.mutable_robot()->mutable_joints()->mutable_position()->CopyFrom(current_positions_);
-  output_pos_.mutable_robot()->mutable_joints()->mutable_position()->CopyFrom(current_positions_);
-
-  for (size_t i = 0; i < hw_states_.size(); ++i) {
-    hw_states_[i] = current_positions_.values(i) / DEG_TO_RAD;
-    hw_commands_[i] = current_positions_.values(i) / DEG_TO_RAD;
-  }
-  RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "System Sucessfully started!");
-
-  return CallbackReturn::SUCCESS;
-}
 
-CallbackReturn ABBSystemPositionOnlyHardware::on_deactivate(
-  const rclcpp_lifecycle::State & /* previous_state */)
-{
-  RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "on_deactivate()");
+  RCLCPP_INFO(LOGGER, "CONNECTED!");
 
-  io_service_.stop();
-  thread_group_.join_all();
+  egm_manager_->read(motion_data_);
 
-  RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "System sucessfully stopped!");
+  RCLCPP_INFO(LOGGER, "System sucessfully started!");
```

Comment:
Tyler says, "don't log the happy path" but I think we can keep this. A little terminal output every once in awhile is a good thing. Let's make it more specific:

```suggestion
  RCLCPP_INFO(LOGGER, "ros2_control hardware interface was successfully started!");
```